Add support for custom resource routes

judgeaxl opened the pull request:
- Not all JSON:API implementation use the exact type name for the resource route. The most common exception is probably using singular type names and plural route names.
- This change allows you to optionally configure a route for each type, which will be used when generating URLs. If no route is specified, it falls back to using the type as before.

# Description

Add a `route` property to `JsonApiResourceOptions` and use it generate resource URLs. Falls back on `type`
 if `route` is undefined.

Fixes #4 

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

# How Has This Been Tested?

There aren't any existing tests that deal with custom routes, so for now I have only tested against a private server. Would be happy to discuss how to incorporate tests.

## Checklist before requesting a review

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

judgeaxl committed:
Add support for custom resource routes

- Not all JSON:API implementation use the exact type name for the
resource route. The most common exception is probably using singular
type names and plural route names.
- This change allows you to optionally configure a route for each type,
which will be used when generating URLs. If no route is specified, it
falls back to using the type as before.

dpimonov commented:
Hey, thanks for that improvement and following style guidelines. I have nothing to add or change. Just add a small note about new option in the documentation here and we can merge changes.

![image](https://user-images.githubusercontent.com/15626139/224843033-56f10bcc-4e0c-43a9-be04-753f36a3eb9a.png)

Then I will trigger a new release.

About tests. I use JsonApi.NET for testing this library. There are `backend` and `spec` folders in the repository root for tests. Backend folder is created to setup different servers. Spec folder contains tests against concrete servers. It might be not so convinient for you to install a lot of stuff and also learning how to configure it as you are using Next.js. We can add test for this option later as there are no big changes involved.

judgeaxl committed:
Add doc for now config option

- I took the liberty of changing the `type` in the example to be
singular, to better illustrate the difference between route and type
here.

judgeaxl commented:
Have you considered mocking the server responses for the tests? You could just run the tests once and capture all traffic. That way it'd require less setup and support tests across a larger variety of possible responses, like server errors, etc.

dpimonov commented:
I thought about it but I really like to test against real implementations due to filtering, sorting, pagination, custom metadata and basically dynamic nature of entity store package. Besides it allows to test more complex use cases. In long perspective it is much easier to maintain when amount of tests grows. Besides due to JSON:API spec there might be tons of different server implementations. Ideally would be to test all of them when there will be more visitors included. Running all test servers can be done with Docker. This will make tests easier but this is something for the future.
